lib/mongoose_push/logger/json.ex

```diff
@@ -0,0 +1,41 @@
+defmodule MongoosePush.Logger.JSON do
```

Comment:
I just wonder why we create new JSON formater if there is several available. I may be missing something.

Reply:
There are some json _backends_ for logger (like [logger_json](https://github.com/Nebo15/logger_json)). If we already have a formatter module for the `loggerfmt` I thought this would work better structure-wise. But we can use this library instead.

Reply:
I'm honestly unsure about this library. It's not a `formatter`, it's a `backend`. I skimmed through their code quickly and I don't understand why they want this way. This is concerning to me, as backeds have to deal with IO, log throttling etc. and even if their implementation is "OK", I imagine that `console` backend that is part of Elixir standard lib is better maintained, and more importantly, better tested. They also have formatter, yes, but I'm not sure if it's compatible with `console` backend, and it's very, very simple one anyway - I'm not a fan of including external dependencies to avoid writing few lines of trivial code ;)
wdyt @michalwski @zofpolkowska ?

Reply:
Yeah, with that feeling I did it the way I did it. Didn't understand the point of having backend as a formatter:)

Reply:
@zofpolkowska Elixir's Logger have pluggable logging `backends` - those are `gen_event` based implementations that are suppose to handle all `Logger.log` calls in some way (write then to IO-device somehow). They need to throttle if IO is too slow, drop logs, do quite a few things to keep the system running properly when there is too much logs compared to IO performance. For example, there may be a file-based `backend` that writes to files. The one default `backend` is called `console` and it's just printing logs to, well, console. custom `formatter` is a configuration option for `console` backend specifically, this is not a global option for `Logger` itself. The `console` backend allows adding custom `formatters` that format data before it's printed on stdout.
Because of this architecture, with assumption that you don't need other backends then `console` (and I think you don't, considering Docker prefers stdout-logging), you can easily get away with just providing custom `formatter`, without need to code the entire logging `backend`. `logger_json` library however implements entire `backend` (also with support for `formatters` as they have two different JSON formatters, one "plain", second compatible with Google Cloud). I don't know why they did that, maybe they wanted generic IO-device to also log to files and still keep formatters. It's important however to make this distinction, because writing a `formatter` to existing `backend` is trivial and it's hard to break anything. Writing entire `backend` however, well, that's totally different story. 

I hope this clarifies things ;)